File: misanuncios/src/main/java/com/yaagoub/misanuncios/application/repository/ImageRepository.java
package com.yaagoub.misanuncios.application.repository;

import com.yaagoub.misanuncios.domain.Image;

public interface ImageRepository
{
    Iterable<Image> getImagesByProduct(long idProduct);
    Image save(Image image);
    void delete(Image image);

    Image findById(long id);


}

File: misanuncios-ui/src/app/application/repositories/image.repository.ts
import { Observable } from "rxjs";
import { Image } from "../../domain/model/image.model";




export abstract class ImageRepository {
    abstract save(image:string,product_id:number): Observable<Image>;
    abstract delete(id: number): Observable<boolean>;
    abstract findByProduct(idProduct:number): Observable<Image[]>;
}
